### PR TITLE
Computer Vision Usecase 5 

### PR DESCRIPTION
**Pull request by ramadevimiriyala**
Overview

This PR relocates Use Case 5 (DAWN: traffic density & haze estimation) into the AI_IOT_Team/UC5_DAWN folder and finalises a reproducible end-to-end pipeline:
dataset → YOLOv8 training → feature extraction → augmentation → correlation analysis.

It now includes:

Extended validation with augmented weather-condition images (fog, rain, sand, snow).

A new correlation plotting script for quick visualization.

Improved evaluation metrics.

Refreshed documentation for easier reproduction.

📂 What’s Included
Data Config

data/dawn.yaml — YOLOv8 dataset split config (train/val/test + class names)

hyp/uc5_strong_aug.yaml — custom hyperparameters with stronger haze/illumination augmentations

⚙️ Pipeline Scripts

prepare_dawn.py — prepare dataset for YOLO format

count_density.py — compute per-image vehicle counts from YOLO labels

haze_score_dcp.py — compute haze scores using Dark Channel Prior

merge_features.py — join vehicle + haze into a single CSV

analyze_uc5.py — correlation, scatter plot, summary metrics

train_uc5.py — training entry point (YOLOv8 with custom hyp)

uc5_demo.py — quick demo inference on validation image

augment_uc5.py — augmentation pipeline using Albumentations (fog, rain, sand, snow)

plot_correlation.py — new script for scatter plot of Vehicle Density vs. Haze by condition

📊 Results

results/uc5_features.csv — merged features

results/uc5_features_with_conditions.csv — merged features with weather-condition labels

results/condition_counts.csv — summary of image/object counts per condition

results/haze_vs_density.png — correlation plot

results/SUMMARY.md — latest validation notes (923 images)

📘 Documentation

Updated README.md with reproduction steps and environment notes

.gitignore excludes runs/, venv/, checkpoints, and heavy weights

🔑 Key Changes

Added augment_uc5.py for external augmentation (fog, rain, sand, snow)

Added plot_correlation.py for visualization of Vehicle Density vs. Haze correlation

Expanded validation set: 154 → 923 images (augmented + original)

Improved metrics: mAP@0.50 = 0.90 (↑ from 0.857)

Updated SUMMARY.md with extended results

Clean repo structure (no heavy weights committed)

Documentation refreshed with step-by-step usage

📈 Metrics (DAWN Validation Set – Extended 923 Images)

mAP@0.50: 0.900 (~90%)

mAP@0.50–0.95: 0.623 (~62%)

Precision: 0.892

Recall: 0.803

Dataset size: 923 validation images (augmented + original)

🛠️ Reproduction (from repo root)
# 1. Train YOLOv8 with augmentations
python src/train_uc5.py  

# 2. Validate best weights
yolo task=detect mode=val \
     model=runs/detect/train6/weights/best.pt \
     data=artificial-intelligence/AI_IOT_Team/UC5_DAWN/data/dawn.yaml  

# 3. Extract features
python src/count_density.py
python src/haze_score_dcp.py
python src/merge_features.py  

# 4. Analyze correlation
python src/analyze_uc5.py
python src/plot_correlation.py  

# 5. Demo inference
python src/uc5_demo.py

### CODE REVIEW

**Review by TKaushik98 — APPROVED** [review]
Good

**Review by caseyjmm — APPROVED** [review]
